test/core/exchange-wrappers/zeroExExchangeWrapper.spec.ts

```diff
@@ -1,39 +1,323 @@
+        secondZeroExOrderTakerAssetAmount = ether(10);
+        secondZeroExOrder = Utils.generateZeroExOrder(
+          NULL_ADDRESS,
+          secondZeroExOrderMakerAccount,
+          NULL_ADDRESS,
+          ZERO,
+          ZERO,
+          secondZeroExOrderMakerAssetAmount,
+          secondZeroExOrderTakerAssetAmount,
+          secondZeroExOrderMakerToken.address,
+          zeroExOrderTakerToken.address,
+          Utils.generateSalt(),
+          TestUtils.ZERO_EX_EXCHANGE_ADDRESS,
+          NULL_ADDRESS,
+          Utils.generateTimestamp(10),
+        );
+
+        const zeroExOrderFillAmount = takerAssetAmount;
+        const zeroExOrderSignature = await utils.signZeroExOrderAsync(secondZeroExOrder);
+        const secondZeroExExchangeWrapperOrder = Utils.generateZeroExExchangeWrapperOrder(
+          secondZeroExOrder,
+          zeroExOrderSignature,
+          zeroExOrderFillAmount
+        );
+        subjectOrderData = Utils.concatBytes([zeroExExchangeWrapperOrder, secondZeroExExchangeWrapperOrder]);
+      });
+
+      it('should receipt the correct amounts of taker tokens and set allowances on ZeroEx/Set proxies', async () => {
+        await subject();
+
+        // First maker token
+        const firstTakerTokenBalance = await zeroExOrderMakerToken.balanceOf.callAsync(zeroExExchangeWrapper.address);
+        expect(firstTakerTokenBalance).to.bignumber.equal(makerAssetAmount);
+        const zeroExMakerTokenAllowance = await zeroExOrderMakerToken.allowance.callAsync(
+          zeroExExchangeWrapper.address,
+          transferProxy.address
+        );
+        expect(zeroExMakerTokenAllowance).to.bignumber.equal(UNLIMITED_ALLOWANCE_IN_BASE_UNITS);
+
+        // Second maker token
+        const secondTakerTokenBalance = await secondZeroExOrderMakerToken.balanceOf.callAsync(
+          zeroExExchangeWrapper.address
+        );
+        expect(secondTakerTokenBalance).to.bignumber.equal(secondZeroExOrderMakerAssetAmount);
+        const secondZeroExMakerTokenAllowance = await secondZeroExOrderMakerToken.allowance.callAsync(
+          zeroExExchangeWrapper.address,
+          transferProxy.address
+        );
+        expect(secondZeroExMakerTokenAllowance).to.bignumber.equal(UNLIMITED_ALLOWANCE_IN_BASE_UNITS);
+
+        // Remaining taker token allowance after two orders
+        const zeroExOrderTakerTokenAllowance = await zeroExOrderTakerToken.allowance.callAsync(
+          zeroExExchangeWrapper.address,
+          TestUtils.ZERO_EX_ERC20_PROXY_ADDRESS
+        );
+        const totalTakerAmountTransferred = takerAssetAmount.add(secondZeroExOrderTakerAssetAmount);
+        const expectedTakerTokenAllowance = UNLIMITED_ALLOWANCE_IN_BASE_UNITS.sub(totalTakerAmountTransferred);
+        expect(zeroExOrderTakerTokenAllowance).to.bignumber.equal(expectedTakerTokenAllowance);
+      });
+    });
+
+    context('when checking the return value', async () => {
+      async function subject(): Promise<any> {
```

Comment:
Do we need to redefine the subject here?

Reply:
Yeah, one is callAsync so we can see the return value. Using `sendTransactionAsync` just returns the transaction hash

Reply:
Ah sure